File: src/main/java/org/apache/fineract/client/services/StaffApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.Call;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.CreateStaffResponse;
import org.apache.fineract.client.models.FormDataContentDisposition;
import org.apache.fineract.client.models.PostStaffRequest;
import org.apache.fineract.client.models.PutStaffRequest;
import org.apache.fineract.client.models.RetrieveOneResponse;
import org.apache.fineract.client.models.UpdateStaffResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface StaffApi {
  /**
   * Create a staff member
   * Creates a staff member.  Mandatory Fields:  officeId, firstname, lastname  Optional Fields:  isLoanOfficer, isActive
   * @param postStaffRequest  (required)
   * @return Call&lt;CreateStaffResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("staff")
  Call<CreateStaffResponse> create3(
    @retrofit2.http.Body PostStaffRequest postStaffRequest
  );

  /**
   * 
   * 
   * @param officeId  (optional)
   * @param dateFormat  (optional)
   * @return Call&lt;Void&gt;
   */
  @GET("staff/downloadtemplate")
  Call<Void> getTemplate1(
    @retrofit2.http.Query("officeId") Long officeId, @retrofit2.http.Query("dateFormat") String dateFormat
  );

  /**
   * 
   * 
   * @param file  (optional)
   * @param locale  (optional)
   * @param dateFormat  (optional)
   * @return Call&lt;String&gt;
   */
  @retrofit2.http.Multipart
  @POST("staff/uploadtemplate")
  Call<String> postTemplate(
    @retrofit2.http.Part("file") FormDataContentDisposition file, @retrofit2.http.Part("locale") String locale, @retrofit2.http.Part("dateFormat") String dateFormat
  );

  /**
   * Retrieve Staff
   * Returns the list of staff members.  Example Requests:  staff     Retrieve a Staff by status  Returns the details of a Staff based on status.  By default it Returns all the ACTIVE Staff.  If status&#x3D;INACTIVE, then it returns all INACTIVE Staff.  and for status&#x3D;ALL, it Returns both ACTIVE and INACTIVE Staff.  Example Requests:  staff?status&#x3D;active
   * @param officeId officeId (optional)
   * @param staffInOfficeHierarchy staffInOfficeHierarchy (optional, default to false)
   * @param loanOfficersOnly loanOfficersOnly (optional, default to false)
   * @param status status (optional, default to &quot;active&quot;)
   * @return Call&lt;List&lt;RetrieveOneResponse&gt;&gt;
   */
  @GET("staff")
  Call<List<RetrieveOneResponse>> retrieveAll16(
    @retrofit2.http.Query("officeId") Long officeId, @retrofit2.http.Query("staffInOfficeHierarchy") Boolean staffInOfficeHierarchy, @retrofit2.http.Query("loanOfficersOnly") Boolean loanOfficersOnly, @retrofit2.http.Query("status") String status
  );

  /**
   * Retrieve a Staff Member
   * Returns the details of a Staff Member.  Example Requests:  staff/1
   * @param staffId staffId (required)
   * @return Call&lt;RetrieveOneResponse&gt;
   */
  @GET("staff/{staffId}")
  Call<RetrieveOneResponse> retrieveOne8(
    @retrofit2.http.Path("staffId") Long staffId
  );

  /**
   * Update a Staff Member
   * Updates the details of a staff member.
   * @param staffId staffId (required)
   * @param putStaffRequest  (required)
   * @return Call&lt;UpdateStaffResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("staff/{staffId}")
  Call<UpdateStaffResponse> update7(
    @retrofit2.http.Path("staffId") Long staffId, @retrofit2.http.Body PutStaffRequest putStaffRequest
  );

}

File: src/main/kotlin/org/openapitools/client/apis/StaffApi.kt
package org.openapitools.client.apis

import okhttp3.MultipartBody
import org.openapitools.client.models.CreateStaffResponse
import org.openapitools.client.models.PostStaffRequest
import org.openapitools.client.models.PutStaffRequest
import org.openapitools.client.models.RetrieveOneResponse
import org.openapitools.client.models.UpdateStaffResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path
import retrofit2.http.Query

interface StaffApi {
    /**
     * Create a staff member
     * Creates a staff member.  Mandatory Fields:  officeId, firstname, lastname  Optional Fields:  isLoanOfficer, isActive
     * Responses:
     *  - 200: OK
     *
     * @param postStaffRequest
     * @return [CreateStaffResponse]
     */
    @POST("v1/staff")
    suspend fun create3(@Body postStaffRequest: PostStaffRequest): CreateStaffResponse

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param officeId  (optional)
     * @param dateFormat  (optional)
     * @return [Unit]
     */
    @GET("v1/staff/downloadtemplate")
    suspend fun getTemplate1(
        @Query("officeId") officeId: Long? = null,
        @Query("dateFormat") dateFormat: String? = null
    ): Unit

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param dateFormat  (optional)
     * @param locale  (optional)
     * @param uploadedInputStream  (optional)
     * @return [kotlin.String]
     */
    @Multipart
    @POST("v1/staff/uploadtemplate")
    suspend fun postTemplate(
        @Part("dateFormat") dateFormat: String? = null,
        @Part("locale") locale: String? = null,
        @Part uploadedInputStream: MultipartBody.Part? = null
    ): String

    /**
     * Retrieve Staff
     * Returns the list of staff members.  Example Requests:  staff     Retrieve a Staff by status  Returns the details of a Staff based on status.  By default it Returns all the ACTIVE Staff.  If status&#x3D;INACTIVE, then it returns all INACTIVE Staff.  and for status&#x3D;ALL, it Returns both ACTIVE and INACTIVE Staff.  Example Requests:  staff?status&#x3D;active
     * Responses:
     *  - 200: OK
     *
     * @param officeId officeId (optional)
     * @param staffInOfficeHierarchy staffInOfficeHierarchy (optional, default to false)
     * @param loanOfficersOnly loanOfficersOnly (optional, default to false)
     * @param status status (optional, default to "active")
     * @return [kotlin.collections.List<RetrieveOneResponse]
     */
    @GET("v1/staff")
    suspend fun retrieveAll16(
        @Query("officeId") officeId: Long? = null,
        @Query("staffInOfficeHierarchy") staffInOfficeHierarchy: Boolean? = false,
        @Query("loanOfficersOnly") loanOfficersOnly: Boolean? = false,
        @Query("status") status: String? = "active"
    ): List<RetrieveOneResponse>

    /**
     * Retrieve a Staff Member
     * Returns the details of a Staff Member.  Example Requests:  staff/1
     * Responses:
     *  - 200: OK
     *
     * @param staffId staffId
     * @return [RetrieveOneResponse]
     */
    @GET("v1/staff/{staffId}")
    suspend fun retrieveOne8(@Path("staffId") staffId: Long): RetrieveOneResponse

    /**
     * Update a Staff Member
     * Updates the details of a staff member.
     * Responses:
     *  - 200: OK
     *
     * @param staffId staffId
     * @param putStaffRequest
     * @return [UpdateStaffResponse]
     */
    @PUT("v1/staff/{staffId}")
    suspend fun update7(
        @Path("staffId") staffId: Long,
        @Body putStaffRequest: PutStaffRequest
    ): UpdateStaffResponse

}
